src/client/Microsoft.Identity.Client/Http/Retry/RegionDiscoveryRetryPolicy.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
+
+namespace Microsoft.Identity.Client.Http.Retry
+{
+    internal class RegionDiscoveryRetryPolicy : IRetryPolicy
```

Comment:
There's too much code duplication between this class and `ImdsRetryPolicy` (and in general between retry policies)

Reply:
I agree with you, but the only difference between `ImdsRetryPolicy` and `RegionDiscoveryRetryPolicy` is a big one:

`ImdsRetryPolicy` does linear OR exponential retry depending on the status code, and `RegionDiscoveryRetryPolicy` ONLY does exponential retry.

Importantly, they both retry on 410 - `ImdsRetryPolicy` retries it linearly and `RegionDiscoveryRetryPolicy` retries it exponentially.